validator/api/src/main/java/tech/pegasys/teku/validator/api/KeyStoreFilesLocator.java

```diff
@@ -80,12 +82,29 @@ private void parseEntry(final String keyFileName, final String passwordFileName)
               passwordFileName));
     }
     if (keyFile.isFile()) {
-      pathMap.putIfAbsent(keyFile.toPath(), passwordFile.toPath());
+      if (!isDepositDataFile(keyFile)) pathMap.putIfAbsent(keyFile.toPath(), passwordFile.toPath());
     } else {
       parseDirectory(keyFile, passwordFile);
     }
   }
 
+  private boolean isDepositDataFile(final File keyFile) {
+    try {
+      byte[] fileData = Files.readAllBytes(keyFile.toPath());
+      ObjectMapper objectMapper = new ObjectMapper();
+      List<Map<String, String>> listData =
+          objectMapper.readValue(fileData, new TypeReference<List<Map<String, String>>>() {});
+
+      if (!listData.isEmpty() && listData.size() > 0) {
+        if (listData.get(0).containsKey("deposit_message_root")
+            && listData.get(0).containsKey("deposit_data_root")) return true;
+      }
```

Comment:
Can probably simplify this to just a single `return` statement and not need the `if` statements (the `&&` operator short circuits so won't execute the right hand side if the left hand side returns false).